dattri/algorithm/rps.py

```diff
@@ -0,0 +1,181 @@
+"""This module implement the representer point selection."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Callable
+
+    from torch import Tensor
+    from torch.utils.data import DataLoader
+
+import warnings
+
+import torch
+from torch.nn.functional import normalize
+
+from dattri.algorithm.utils import (
+    _check_shuffle,
+    get_rps_weight,
+    rps_finetune_theta,
+)
+from dattri.model_utils.hook import get_final_layer_io
+
+from .base import BaseAttributor
+
+
+class RPSAttributor(BaseAttributor):
+    """Representer point selection attributor."""
+
+    def __init__(
+        self,
+        loss_func: Callable,
+        model: torch.nn.Module,
+        final_linear_layer_name: str,
+        nomralize_preactivate: bool = False,
+        l2_strength: float = 0.003,
+        epoch: int = 3000,
+        device: str = "cpu",
+    ) -> None:
+        """Representer point selection attributor.
+
+        Args:
+            loss_func (Callable): The loss function to be attributed. The inputs are
+                list of pre-activation values (f_i in the paper) and list of labels.
+                Typical examples are BCELoss and CELoss.
```

Comment:
Still Called it a `target_func` but add some additional comments to state this is typically a loss function.